src/Volume.ts

```diff
@@ -244,19 +246,32 @@ export default class Volume {
     let noReload =
       this.loadSpec.time === this.loadSpecRequired.time &&
       this.loadSpec.subregion.containsBox(this.loadSpecRequired.subregion) &&
-      this.loadSpecRequired.channels.every((channel) => this.loadSpec.channels.includes(channel));
+      this.loadSpecRequired.channels.every((channel) => this.loadSpec.channels.includes(channel)) &&
+      this.loadSpecRequired.scaleLevelBias === this.loadSpec.scaleLevelBias;
 
     // An update to `subregion` should trigger a reload when the new subregion is not contained in the old one
     // OR when the new subregion is smaller than the old one by enough that we can load a higher scale level.
-    if (noReload && !this.loadSpec.subregion.equals(this.loadSpecRequired.subregion)) {
+    if (
+      noReload &&
+      (!this.loadSpec.subregion.equals(this.loadSpecRequired.subregion) ||
+        this.loadSpecRequired.maxAtlasEdge !== this.loadSpec.maxAtlasEdge ||
+        this.loadSpecRequired.multiscaleLevel !== this.loadSpec.multiscaleLevel)
+    ) {
       const currentScale = this.imageInfo.multiscaleLevel;
-      // `LoadSpec.multiscaleLevel`, if specified, forces a cap on the scale level we can load.
-      const minLevel = this.loadSpec.multiscaleLevel ?? 0;
       // Loaders should cache loaded dimensions so that this call blocks no more than once per valid `LoadSpec`.
       const dims = await this.loader?.loadDims(this.loadSpecRequired);
       if (dims) {
-        const loadableLevel = estimateLevelForAtlas(dims.map(({ shape }) => [shape[2], shape[3], shape[4]]));
-        noReload = currentScale <= Math.max(loadableLevel, minLevel);
+        // `LoadSpec.scaleLevelBias`, if specified, nudges the scale level we pick up or down.
+        const levelBias = this.loadSpecRequired.scaleLevelBias ?? 0;
+        // `LoadSpec.multiscaleLevel`, if specified, forces a cap on the scale level we can load.
+        const minLevel = this.loadSpecRequired.multiscaleLevel ?? 0;
+
+        const dimsZYX = dims.map(({ shape }): [number, number, number] => [shape[2], shape[3], shape[4]]);
+        const optimalLevel = estimateLevelForAtlas(dimsZYX, this.loadSpecRequired.maxAtlasEdge);
+
+        const levelToLoad = Math.max(optimalLevel + levelBias, minLevel);
+        const clampedLevelToLoad = Math.max(0, Math.min(dims.length - 1, levelToLoad));
+        noReload = currentScale === clampedLevelToLoad;
```

Comment:
now that you calculated clampedLevelToLoad, I am not seeing where it's being used to actually load anything.

Reply:
I think I see the same computation in zarr_utils... maybe they could be factored into a simple utils function.

Reply:
reorganized where this kind of scale level resolution happens pretty significantly, to hopefully be more general